File: src/main/java/de/omilke/bankingfx/resources/ImageProvider.java
package de.omilke.bankingfx.resources;

import javafx.scene.image.Image;

/**
 * Created by olli on 27.11.2015.
 */
public class ImageProvider {

    public static Image readImageFromMetaInf(ImageType type) {

        return new Image(ImageProvider.class.getClassLoader().getResourceAsStream("META-INF/img/" + type.getValue()));
    }
}

File: src/main/java/de/omilke/bankingfx/resources/ImageProvider.kt
package de.omilke.bankingfx.resources

import javafx.scene.image.Image

/**
 * Created by olli on 27.11.2015.
 */
object ImageProvider {

    fun readImageFromMetaInf(type: ImageType): Image {

        return Image(ImageProvider::class.java.classLoader.getResourceAsStream("META-INF/img/" + type.value))
    }
}